### PR TITLE
chore(js): add rudamentary types for experiments

### PR DESCRIPTION
**Pull request by mikeldking**
Adds types under client for now to get the abstractions right under the client.

### CODE REVIEW

File: js/packages/phoenix-client/src/types/experiments.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+   * This is null if the trace is deleted or never recorded
+   */
+  trace_id: string | null;
+}
+
+export type TaskOutput = string | boolean | number | object | null;
+
+export type ExperimentTask =
+  | ((example: Example) => Promise<TaskOutput>)
+  | ((example: Example) => TaskOutput);
```
Comment (cephalization) [inline js/packages/phoenix-client/src/types/experiments.ts:65]: Could task output just be generic on experiment task?

File: js/packages/phoenix-client/src/types/experiments.ts [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  metadata: Record<string, unknown>;
+  explanation: string | null;
+};
+
+export interface ExperimentEvaluationRun extends Node {
+  experimentRunId: string;
+  startTime: Date;
+  endTime: Date;
+  /**
+   * THe name of the evaluation
```
Comment (cephalization) [inline js/packages/phoenix-client/src/types/experiments.ts:48]:
```suggestion
   * The name of the evaluation
```

File: js/packages/phoenix-client/src/types/experiments.ts [hunk 3]
```diff
@@ -0,0 +1,77 @@
+import { AnnotatorKind } from "./annotations";
+import { Node } from "./core";
+import { Example } from "./datasets";
+/**
```
Comment (cephalization) [inline js/packages/phoenix-client/src/types/experiments.ts:4]: Is prettier configured? Should be a gap here

**Review by cephalization — APPROVED** [review]
Minor style nits

File: js/packages/phoenix-client/src/types/experiments.ts (hunk 1 above)
Comment (mikeldking) [reply js/packages/phoenix-client/src/types/experiments.ts:65]: The example maybe, but not the output right now. This type is not 100% correct but a decent place to start.

File: js/packages/phoenix-client/src/types/datasets.ts [hunk 4]
```diff
@@ -0,0 +1,21 @@
+import { Node } from "./core";
+
+/**
+ * An example is a record to feed into an AI task
+ */
+export interface Example extends Node {
+  id: string;
+  updatedAt: Date;
```
Comment (Parker-Stafford) [inline js/packages/phoenix-client/src/types/datasets.ts:8]: should this be nullable? thinking from perspective of ad hoc datasets

Comment (mikeldking) [reply js/packages/phoenix-client/src/types/datasets.ts:8]: Yeah we can relax as we go

File: js/packages/phoenix-client/src/types/datasets.ts [hunk 5]
```diff
@@ -0,0 +1,21 @@
+import { Node } from "./core";
+
+/**
+ * An example is a record to feed into an AI task
+ */
+export interface Example extends Node {
+  id: string;
```
Comment (Parker-Stafford) [inline js/packages/phoenix-client/src/types/datasets.ts:7]: not needed cause it extends node, same with dataset below

File: js/packages/phoenix-client/src/types/experiments.ts [hunk 6]
```diff
@@ -0,0 +1,78 @@
+import { AnnotatorKind } from "./annotations";
+import { Node } from "./core";
+import { Example } from "./datasets";
+
+/**
+ * An experiment is a set of task runs on a dataset version
+ */
+export interface Experiment extends Node {
+  id: string;
```
Comment (Parker-Stafford) [inline js/packages/phoenix-client/src/types/experiments.ts:9]: same here, although fine to keep if you want

File: js/packages/phoenix-client/src/types/experiments.ts [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+  startTime: Date;
+  endTime: Date;
+  /**
+   * What experiment the run belongs to
+   */
+  experimentId: string;
+  datasetExampleId: string;
+  repetitionNumber: number;
+  output?: string | {} | null;
```
Comment (Parker-Stafford) [inline js/packages/phoenix-client/src/types/experiments.ts:32]: I don't think this is right {} is like an empty object i think maybe:

```suggestion
  output?: string | Record<string, unknown> | null;
```
or more defensively
```suggestion
  output?: string | Record<unknown, unknown> | null;
```

File: js/packages/phoenix-client/src/types/experiments.ts [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+  error: string | null;
+  traceId: string | null;
+}
+
+export type EvaluationResult = {
+  score: number | null;
+  label: string | null;
+  metadata: Record<string, unknown>;
+  explanation: string | null;
```
Comment (Parker-Stafford) [inline js/packages/phoenix-client/src/types/experiments.ts:41]: good to make these unsettable as well?

```suggestion
  score?: number | null;
  label?: string | null;
  metadata?: Record<string, unknown>;
  explanation?: string | null;
```

File: js/packages/phoenix-client/src/types/datasets.ts (hunk 4 above)
Comment (mikeldking) [reply js/packages/phoenix-client/src/types/datasets.ts:8]: I'm just matching the python right now

File: js/packages/phoenix-client/src/types/experiments.ts [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+  experimentRunId: string;
+  startTime: Date;
+  endTime: Date;
+  /**
+   * THe name of the evaluation
+   */
+  name: string;
+  annotatorKind: AnnotatorKind;
+  error: string | null;
+  result: EvaluationResult | null;
```
Comment (Parker-Stafford) [inline js/packages/phoenix-client/src/types/experiments.ts:52]: same here re unsettable

```suggestion
  error?: string | null;
  result?: EvaluationResult | null;
```

File: js/packages/phoenix-client/src/types/datasets.ts (hunk 5 above)
Comment (mikeldking) [reply js/packages/phoenix-client/src/types/datasets.ts:7]: 👍🏻

File: js/packages/phoenix-client/src/types/experiments.ts (hunk 8 above)
Comment (mikeldking) [reply js/packages/phoenix-client/src/types/experiments.ts:39]: I think so - but let's relax as we go.

File: js/packages/phoenix-client/src/types/experiments.ts (hunk 7 above)
Comment (mikeldking) [reply js/packages/phoenix-client/src/types/experiments.ts:30]: It's technically anything that can be JSON dumped. I'll commit the former for now

**Pull request merged**